src/buffer.rs

```diff
@@ -1569,6 +1569,12 @@ impl UnicodeBuffer {
         self.0.push_str(str);
     }
 
+    /// Appends a character to a buffer with the given cluster value.
+    #[inline]
+    pub fn add(&mut self, codepoint: char, cluster: u32) {
+        self.0.add(codepoint as u32, cluster)
```

Comment:
You have to call 

```
self.0.context_len[1] = 0;
```

as well.

https://github.com/harfbuzz/harfbuzz/blob/630f09c8b6022184f7a414bb5e07c01898fae60c/src/hb-buffer.cc#L1355